docs_src/processor/tutorial001.py

```diff
@@ -0,0 +1,41 @@
+import asyncio
+
+from fastcrawler import BaseModel, Depends, Process, Spider, XPATHField
+from fastcrawler.schedule import ProcessController, RocketryApplication
+
+
+class PersonData(BaseModel):
+    name: str = XPATHField(query="//td[1]", extract="text")
+    age: int = XPATHField(query="//td[2]", extract="text")
+
+
+class PersonPage(BaseModel):
+    person: list[PersonData] = XPATHField(query="//table//tr", many=True)
+
+
+async def get_urls():
+    return {f"http://localhost:8000/persons/{id}" for id in range(20)}
+
+
+class MySpider(Spider):
+    engine_request_limit = 20
+    data_model = PersonPage
+    start_url = Depends(get_urls)
+
+    async def save(self, all_data: list[PersonPage]):
+        assert all_data is not None
+        assert len(all_data) == 20
+
+
+async def main():
+    process = Process(
+        spider=MySpider(),
+        cond="every 1 second",
+        controller=ProcessController(app=RocketryApplication()),
+    )
+    await process.add_spiders()
```

Comment:
add_spider() -> empty?!
where is the spider to add?

Reply:
we define them in the chain
```python
async def main():
    process = Process(
        spider=MySpider() >> MySpiderTwo() >> MySpiderThree(),
        cond="every 1 second",
        controller=ProcessController(app=RocketryApplication()),
    )
    await process.add_spiders()
 ```
the method `add_spiders ` also adds them to the controller, if there's any defined.
it is not meant to be called like this, usually this happen underhood in FastCrawler App. but if a user want to initiate and run a process manually, that's the way to go.